src/hooks/useGlobalAppState.ts

```diff
@@ -75,51 +75,63 @@ const onStateChange = (
  * Do NOT use it anywhere else than in the <App /> component,
  * for it could create unintended side effects.
  */
-export const useGlobalAppState = (): void => {
-  const client = useClient()
-  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-  const routeIndex = useNavigationState(state => state?.index)
-  const { sharingIntentStatus } = useSharingState()
-
-  // On app start
-  useEffect(() => {
-    const appStart = async (): Promise<void> => {
-      if (await getIsSecurityFlowPassed()) {
-        log.info('useGlobalAppState: app start, security flow passed')
 
-        if (sharingIntentStatus === SharingIntentStatus.OpenedViaSharing) {
-          log.info('useGlobalAppState: app start, sharing mode')
-          navigate(routes.sharing)
-        } else {
-          log.info('useGlobalAppState: app start, not sharing mode')
-        }
-      } else {
-        log.info('useGlobalAppState: app start, security flow not passed')
-      }
-    }
+interface GlobalAppStateProps {
+  onNavigationRequest: (route: string) => void
+}
 
-    log.info('useGlobalAppState: app start')
-    void appStart()
-  }, [routeIndex, sharingIntentStatus])
+export const useGlobalAppState = ({
+  onNavigationRequest
+}: GlobalAppStateProps): void => {
+  // Ref to track if the logic has already been executed
+  const hasExecuted = useRef(false)
+  const client = useClient()
+  const { sharingIntentStatus, errored } = useSharingState()
 
   useEffect(() => {
     let subscription: NativeEventSubscription | undefined
 
     // If there's no client, we don't need to listen to app state changes
     // because we can't lock the app anyway
-    if (client && !subscription) {
+    if (!hasExecuted.current && client && !subscription) {
```

Comment:
Does this mean that if `client` changes (or any other array dependency) the subscription will be removed but will never register again (as `hasExecuted.current` is never set back to `false`)? Is that expected?

Reply:
I will double-check the lifecycle of this event listener, I didn't notice a removal that broke my appStatus change flow but it's better to be safe